### PR TITLE
Small features

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
Merge as separate commits.

### CODE REVIEW

**Comment by lkwg82** [comment]
LGTM

File: cmd/syncthing/monitor.go [hunk 1]
```diff
@@ -202,8 +202,24 @@ func copyStderr(stderr io.Reader, dst io.Writer) {
...
 
 				l.Warnf("Panic detected, writing to \"%s\"", panicFd.Name())
-				l.Warnln("Please check for existing issues with similar panic message at https://github.com/syncthing/syncthing/issues/")
-				l.Warnln("If no issue with similar panic message exists, please create a new issue with the panic log attached")
+				if strings.Contains(line, "leveldb") && strings.Contains(line, "corrupt") {
+					l.Warnln(`
+*********************************************************************************
+* Crash due to corrupt database.                                                *
+*                                                                               *
+* This crash usually occours due to one of the following reasons:               *
```
Comment (calmh) [inline cmd/syncthing/monitor.go]: "occurs"

File: lib/config/optionsconfiguration.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ type OptionsConfiguration struct {
 	OverwriteRemoteDevNames bool     `xml:"overwriteRemoteDeviceNamesOnConnect" json:"overwriteRemoteDeviceNamesOnConnect" default:"false"`
 	TempIndexMinBlocks      int      `xml:"tempIndexMinBlocks" json:"tempIndexMinBlocks" default:"10"`
 	UnackedNotificationIDs  []string `xml:"unackedNotificationID" json:"unackedNotificationIDs"`
+	TrafficClass            int      `xml:"trafficClass" json:"trafficClass" default:"0"`
```
Comment (calmh) [inline lib/config/optionsconfiguration.go]: Unnecessary "default" tag but whatever

File: lib/connections/tcp_dial.go [hunk 3]
```diff
@@ -39,6 +39,16 @@ func (d *tcpDialer) Dial(id protocol.DeviceID, uri *url.URL) (internalConn, erro
 		return internalConn{}, err
 	}
 
+	err = dialer.SetTCPOptions(conn)
```
Comment (calmh) [inline lib/connections/tcp_dial.go]: Good catch

File: lib/dialer/public.go [hunk 4]
```diff
@@ -75,3 +78,22 @@ func SetTCPOptions(conn net.Conn) error {
 		return fmt.Errorf("unknown connection type %T", conn)
 	}
 }
+
+func SetTrafficClass(conn net.Conn, class int) error {
+	switch conn := conn.(type) {
+	case *net.TCPConn:
+		e1 := ipv4.NewConn(conn).SetTOS(class)
+		e2 := ipv6.NewConn(conn).SetTrafficClass(class)
```
Comment (calmh) [inline lib/dialer/public.go]: I guess one of these will always fail as the connection is either one or the other?

Comment (AudriusButkevicius) [inline lib/dialer/public.go]: Yeah, but nothing fatals out because of this.

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
@AudriusButkevicius: Build status is `pending`. I'll wait until it goes green and then merge!

**Comment by calmh** [comment]
No good to let st-review merge this into a single mystery commit

**Comment by AudriusButkevicius** [comment]
baahh

**Comment by st-review** [comment]
@AudriusButkevicius: Merge failed:

```
$ git fetch -f origin refs/pull/3830/head:pr-3830
$ git fetch -f origin master:orig/master
From github.com:syncthing/syncthing
   2d47242..ba17cc0  master     -> orig/master
   b8c1c0e..ba17cc0  master     -> origin/master
$ git reset --hard
HEAD is now at b8c1c0e cmd/syncthing: Enable better crypto, print negotiated cipher suite
$ git checkout master
Already on 'master'
Your branch is behind 'origin/master' by 8 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
$ git reset --hard orig/master
HEAD is now at ba17cc0 gui: Show introducedBy (fixes #3809)
$ git clean -fxd
$ git merge --squash --no-commit pr-3830
Automatic merge went well; stopped before committing as requested
Squash commit -- not updating HEAD
$ git commit -F -
On branch master
Your branch is up-to-date with 'origin/master'.

nothing to commit, working directory clean

```

**Pull request merged**